src/code_805D8C8_1.c

```diff
@@ -0,0 +1,4835 @@
+"b _0805E1AE\n"
+".align 2, 0\n"
+"_0805E17C: .4byte gRealInputs\n"
+"_0805E180: .4byte gDungeon\n"
+"_0805E184: .4byte 0x0000066c\n"
+"_0805E188:\n"
+"mov r0, r8\n"
+"adds r0, 0x48\n"
+"movs r2, 0x1\n"
+"strb r2, [r0]\n"
+"b _0805E1AE\n"
+"_0805E192:\n"
+"movs r3, 0x1\n"
+"ands r4, r3\n"
+"cmp r4, 0\n"
+"beq _0805E1A0\n"
+"movs r0, 0x23\n"
+"bl sub_803E724\n"
+"_0805E1A0:\n"
+"movs r0, 0xF\n"
+"bl sub_803E46C\n"
+"ldr r5, [sp, 0x14]\n"
+"cmp r5, 0\n"
+"bne _0805E1AE\n"
+"b _0805DB24\n"
+"_0805E1AE:\n"
+"ldr r1, [sp, 0x18]\n"
+"ldrb r0, [r1]\n"
+"cmp r0, 0\n"
+"beq _0805E1BA\n"
+"bl sub_804AA60\n"
+"_0805E1BA:\n"
+"ldr r4, [sp, 0x1C]\n"
+"ldrh r0, [r4]\n"
+"cmp r0, 0x2D\n"
+"beq _0805E1C6\n"
+"cmp r0, 0x13\n"
+"bne _0805E1E0\n"
+"_0805E1C6:\n"
+"mov r0, r10\n"
+"bl HandleTalkFieldAction\n"
+"bl sub_8044B28\n"
+"lsls r0, 24\n"
+"cmp r0, 0\n"
+"bne _0805E2B0\n"
+"movs r0, 0\n"
+"bl sub_8044C50\n"
+"bl _0805D980\n"
+"_0805E1E0:\n"
+"lsls r0, r6, 24\n"
+"lsrs r1, r0, 24\n"
+"cmp r1, 0\n"
+"bne _0805E222\n"
+"ldr r2, _0805E210\n"
+"ldr r0, [r2]\n"
+"ldr r3, _0805E214\n"
+"adds r0, r3\n"
+"strb r1, [r0]\n"
+"ldrh r0, [r4]\n"
+"cmp r0, 0\n"
+"beq _0805E218\n"
+"mov r0, r10\n"
+"movs r1, 0\n"
+"bl IsNotAttacking\n"
+"lsls r0, 24\n"
+"cmp r0, 0\n"
+"bne _0805E2B0\n"
+"movs r0, 0xF\n"
+"bl sub_803E46C\n"
+"b _0805E2B0\n"
+".align 2, 0\n"
+"_0805E210: .4byte gDungeon\n"
+"_0805E214: .4byte 0x0000066d\n"
+"_0805E218:\n"
+"movs r0, 0xF\n"
+"bl sub_803E46C\n"
+"bl _0805D980\n"
+"_0805E222:\n"
+"movs r0, 0xF\n"
+"bl sub_803E46C\n"
+"bl sub_8047158\n"
+"movs r2, 0\n"
+"lsrs r0, r6, 8\n"
+"lsls r0, 24\n"
+"cmp r0, 0\n"
+"bne _0805E238\n"
+"movs r2, 0x1\n"
+"_0805E238:\n"
+"lsrs r1, r6, 16\n"
+"lsls r1, 24\n"
+"lsrs r1, 24\n"
+"adds r0, r2, 0\n"
+"bl ShowFieldMenu\n"
+"bl ResetRepeatTimers\n"
+"bl ResetUnusedInputStruct\n"
+"ldr r1, _0805E294\n"
+"movs r0, 0\n"
+"strb r0, [r1]\n"
+"movs r0, 0\n"
+"ldr r5, [sp, 0x18]\n"
+"strb r0, [r5]\n"
+"bl sub_804AA60\n"
+"bl sub_8044B28\n"
+"lsls r0, 24\n"
+"lsrs r1, r0, 24\n"
+"cmp r1, 0\n"
+"bne _0805E2B0\n"
+"ldrh r0, [r4]\n"
+"cmp r0, 0\n"
+"beq _0805E29C\n"
+"cmp r0, 0x2B\n"
+"bne _0805E27E\n"
+"ldr r2, _0805E298\n"
+"ldr r0, [r2]\n"
+"movs r3, 0x1\n"
+"strb r3, [r0, 0x4]\n"
+"ldr r0, [r2]\n"
+"strb r3, [r0, 0x3]\n"
+"_0805E27E:\n"
+"ldr r5, [sp, 0x1C]\n"
+"ldrh r0, [r5]\n"
+"cmp r0, 0x2E\n"
+"bne _0805E2B0\n"
+"ldr r2, _0805E298\n"
+"ldr r0, [r2]\n"
+"movs r3, 0x1\n"
+"strb r3, [r0, 0x4]\n"
+"ldr r0, [r2]\n"
+"strb r1, [r0, 0x3]\n"
+"b _0805E2B0\n"
+".align 2, 0\n"
+"_0805E294: .4byte gUnknown_202F22D\n"
+"_0805E298: .4byte gDungeon\n"
+"_0805E29C:\n"
+"movs r0, 0xF\n"
+"bl sub_803E46C\n"
+"ldr r5, _0805E2C0\n"
+"ldr r0, [r5]\n"
+"ldrb r0, [r0, 0x4]\n"
+"cmp r0, 0\n"
+"bne _0805E2B0\n"
+"bl _0805D980\n"
+"_0805E2B0:\n"
+"add sp, 0x24\n"
+"pop {r3-r5}\n"
+"mov r8, r3\n"
+"mov r9, r4\n"
+"mov r10, r5\n"
+"pop {r4-r7}\n"
+"pop {r0}\n"
+"bx r0\n"
+".align 2, 0\n"
+"_0805E2C0: .4byte gDungeon\n");
+}
+
+#endif
+
+/* Nope
```

Comment:
Is this the start of the function? If we want to merge this in then let's change this comment to be more descriptive.